applications/FluidDynamicsApplication/custom_utilities/rotating_frame_utility.cpp

```diff
@@ -0,0 +1,110 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:     Sebastian Ares de Parga Regalado
+//
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Application includes
+#include "rotating_frame_utility.h"
+
+namespace Kratos 
+{
+
+void RotatingFrameUtility::ApplyVelocityToRotatingObject(
+    ModelPart& rModelPart,
+    const array_1d<double, 3>& rAxisOfRotation, 
+    const double& rOmega, 
+    const array_1d<double, 3>& rCenterOfRotation)
+{
+    KRATOS_TRY;
+
+    // Creating the angular velocity vector
+    array_1d<double, 3> angular_velocity_vector = rOmega * rAxisOfRotation;
+
+    // Apply the velocity calculations to each node in parallel
+    block_for_each(rModelPart.Nodes(), [&](Node& rNode) {
+        // Getting the current coordinates of the node
+        auto& r_point = rNode.Coordinates();
+
+        // Calculating the position vector (relative to the rotation center)
+        array_1d<double, 3> position_vector = r_point - rCenterOfRotation;
+
+        // Computing the velocity due to rotation (v = omega cross r)
+        array_1d<double, 3> velocity_vector = MathUtils<double>::CrossProduct(angular_velocity_vector, position_vector);
+
+        // Setting the node's velocity
+        rNode.FastGetSolutionStepValue(VELOCITY) = velocity_vector;
+
+        // Fix the velocity components
+        rNode.Fix(VELOCITY_X);
+        rNode.Fix(VELOCITY_Y);
+        rNode.Fix(VELOCITY_Z);
```

Comment:
Not sure if the fixity must be done in here. What if I want to apply this to a problem in which the `VELOCITY` is not a DOF (e.g. convection).